Model/Config/Source/Payment/Settings.php

```diff
@@ -75,6 +80,19 @@ public function getOrderStatus()
         return $this->getDataConfig('payment/settings/order_status');
     }
 
+    public function getNotificationEndpoint()
+    {
+        try {
+            return $this->_storeManager->getStore()->getBaseUrl() . 'snap/payment/notification';
+        } catch (NoSuchEntityException $e) {
+        }
```

Comment:
What will be returned if it comes in to the `catch` block? is it `null`? 
Then please make sure the caller of `getNotificationEndpoint()` to handle `null` case.

Reply:
Just add to catch block with Midtrans logger and return null. 
`
catch (NoSuchEntityException $e) {
            $this->midtransLogger->midtransError('GetNotificationEndpointFunction: ' . $e);
            return null;
        }
`